### PR TITLE
[Raiden Weight Sync 7/7] Clean up rollout path and remove redundant weight sync logic

### PR DESCRIPTION
**Pull request by YixuanWang-99**
## Overview
Part of the stacked Raiden weight-sync enablement PR chain replacing #5089.

**Stack:**
- [M7] https://github.com/AI-Hypercomputer/maxtext/pull/5166: Cross-repo drift guard
- [M1] https://github.com/AI-Hypercomputer/maxtext/pull/5167: Inhomogeneous layer cycle support in raiden_unscan
- [M2] https://github.com/AI-Hypercomputer/maxtext/pull/5168: MoE 128-lane layout and padding (pairs with Tunix T5)
- [M3] https://github.com/AI-Hypercomputer/maxtext/pull/5169: Target-free streaming weight conversion
- [M5] https://github.com/AI-Hypercomputer/maxtext/pull/5170: Shared expert gate weight_dtype fix
- [M4] https://github.com/AI-Hypercomputer/maxtext/pull/5171: TrainingEngine Raiden FFI integration (pairs with Tunix T2a)
- **[M6] AI-Hypercomputer/maxtext (this PR)**: Rollout cleanup

## Details
- Net −260 lines.
- Removes deprecated and redundant rollout and weight sync duplication in `maxtext_vllm_rollout.py`, now that the logic is cleanly owned by M3/M4.
- Remaps hybrid KV cache groups in `adapter.py` using `layer_name_to_kvcache_index` before passing to decoder layers.
- Adds `rollout_tensor_parallelism` field to VLLM config.
- To keep CI happy we also update the google-tunix commit hash in `post_train_github_deps.txt`. This pulls in the latest API changes to the raiden_synchronizer from Tunix that are required for unit tests to pass.
- 
# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run end-to-end tests tests and provided workload links above if applicable.
- [X] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/maxtext/integration/vllm/maxtext_vllm_rollout.py [hunk 1]
```diff
@@ -26,106 +26,21 @@
...
 import json
-import logging
 import re
 import time
 import traceback
 from typing import Any, Optional, Tuple
 
-import jax
-import jax.numpy as jnp
+from absl import logging
```
Comment (SurbhiJainUSC) [inline src/maxtext/integration/vllm/maxtext_vllm_rollout.py]: Why do we need to change `import logging` to `from absl import logging`?

File: src/maxtext/integration/vllm/maxtext_vllm_rollout.py [hunk 2]
```diff
@@ -450,158 +279,68 @@ def unroll_gemma_scanned_weights(weights):
...
   ):
     super().__init__(tokenizer=tokenizer, config=config)
-    self._converter = converter
-    self.converter = converter
     self._direct_maxtext_sync = direct_maxtext_sync
-    self._scan_axis = scan_axis
-    self._layer_pattern_length = layer_pattern_length
+    self._model_name = model_name or ""
+    if not self._model_name:
+      model_config = getattr(config, "model_config", None)
```
Comment (SurbhiJainUSC) [inline src/maxtext/integration/vllm/maxtext_vllm_rollout.py]: Can this logic be simplified?

File: src/maxtext/integration/vllm/maxtext_vllm_rollout.py (hunk 1 above)
Comment (igorts-git) [reply src/maxtext/integration/vllm/maxtext_vllm_rollout.py]: I reverted it back to `import logging`

File: src/maxtext/integration/vllm/maxtext_vllm_rollout.py [hunk 3]
```diff
@@ -450,158 +279,68 @@ def unroll_gemma_scanned_weights(weights):
...
 
 
+def _log_and_flush_traceback(msg: str) -> None:
+  """Logs an error with formatted traceback and flushes all logging handlers."""
+  logging.error("%s:\n%s", msg, traceback.format_exc())
+  for handler in logging.getLogger().handlers:
+    try:
+      handler.flush()
+    except Exception:  # pylint: disable=broad-except
+      pass
```
[the remark this answers is not in the document]
Comment (igorts-git) [reply src/maxtext/integration/vllm/maxtext_vllm_rollout.py:456]: I restored `import logging`

File: src/maxtext/integration/vllm/maxtext_vllm_rollout.py [hunk 4]
```diff
@@ -450,158 +279,68 @@ def unroll_gemma_scanned_weights(weights):
...
+      hf_config = getattr(model_config, "hf_config", None)
+      model_type = getattr(hf_config, "model_type", "") or ""
+      arch_str = " ".join(str(a) for a in architectures)
+      self._is_gemma = (
+          "gemma" in str(self._model_name).lower()
+          or "gemma" in str(model_type).lower()
+          or "gemma" in str(arch_str).lower()
+      )
+    else:
+      self._is_gemma = "gemma" in str(self._model_name).lower()
```
[the remark this answers is not in the document]
Comment (igorts-git) [reply src/maxtext/integration/vllm/maxtext_vllm_rollout.py]: fixed

File: src/maxtext/integration/vllm/maxtext_vllm_rollout.py (hunk 2 above)
Comment (igorts-git) [reply src/maxtext/integration/vllm/maxtext_vllm_rollout.py]: Since there is no `config.model_config` in VllmConfig, I removed and simplified that code.

File: src/maxtext/training_engine/maxtext_engine.py [hunk 5]
```diff
@@ -37,6 +37,11 @@
 from maxtext.common import train_state_nnx
 from maxtext.configs import pyconfig
 from maxtext.integration.tunix.weight_mapping import raiden_unscan
+from maxtext.integration.vllm.convert_utils import (
+    is_verify_weights_enabled,
```
Comment (SurbhiJainUSC) [inline src/maxtext/training_engine/maxtext_engine.py:40]: Do these methods exist?

Comment (YixuanWang-99) [reply src/maxtext/training_engine/maxtext_engine.py:40]: Jetski forget to push it :). Good catch

**Review by igorts-git — APPROVED** [review]

File: src/maxtext/training_engine/maxtext_engine.py [hunk 6]
```diff
@@ -1774,6 +1822,10 @@ def save_checkpoint(self, metadata: Any, **kwargs: Any) -> None:
       metadata: Checkpoint metadata payload from Orchestrator.
       **kwargs: Additional checkpoint saving options.
     """
+    if not getattr(self._config, "enable_checkpointing", True) or not self._checkpoint_dir():
```
Comment (SurbhiJainUSC) [inline src/maxtext/training_engine/maxtext_engine.py]: Do we need this?

Comment (igorts-git) [reply src/maxtext/training_engine/maxtext_engine.py]: I think it is not too bad. The default setting of enable_checkpointing is True. So, it will only take place when the user explicitly asks us to not checkpoint (useful for faster debugging).

**Review by SurbhiJainUSC — APPROVED** [review]
